include/ginkgo/core/base/polymorphic_object.hpp

```diff
@@ -708,14 +772,22 @@ class EnablePolymorphicObject
 
     PolymorphicObject* clear_impl() override
     {
-        return polymorphic_object_traits<ConcreteObject>::clear_impl(self());
+        *self() =
+            ConcreteObject{self()->get_executor(), self()->get_communicator()};
+        return this;
     }
 
 private:
     GKO_ENABLE_SELF(ConcreteObject);
 };
 
 
+}  // namespace experimental
+
+
+#endif
```

Comment:
nit: For this `endif`, it is useful to know which `#if` it corresponds to.
Please also add it to all other `#if GINKGO_BUILD_MPI` `#endif`s.
```suggestion
#endif  // GINKGO_BUILD_MPI
```

Reply:
I always think that clang-format will add this, but it seems it doesn't, so I forget.